### ISSUE TITLE
[bug]: Firefox Win 11 DOMException: Node.removeChild: The node to be removed is not a child of this node

### ISSUE THREAD

**Issue by happytomatoe** [nand2tetris/web-ide#425]
### Tool

Hardware Simulator

### Interface

Website (https://nand2tetris.github.io/web-ide)

### What happened?

When using firefox 129.0 (64-bit) on Win 11 when I go to https://nand2tetris.github.io/web-ide/chip and try to run a test I get the the following
![Screenshot 2024-08-13 181152](https://github.com/user-attachments/assets/1c4b09a3-8aa3-4025-9fe8-e7ef3b58aa7e)
When I try to open "Converter tool" I see an empty page. Console shows same/similar error. When using Linux fedora 40 or chrome everything LGTM. Is this expected? What browsers/platform does the web ide support?
    

### Do you want to try to fix this bug?

- [ ] I want to try to add this feature!

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by happytomatoe** [comment]
Sorry, the tool should be Hardware simulator.

**Comment by netalondon** [comment]
Hi @happytomatoe, Thanks for the bug report.

I apologize for the late reply, but I just got to testing this and can't replicate your issue.
Is this still happening?

**Comment by happytomatoe** [comment]
@netalondon Let me try again. BTW Do you have plans to include support for firefox? I guess that would require a lot of effort. Currently I am writing a Jack game. I've tried to run it inside VM Emulator using firefox and it's lagging a lot compared to when using Chrome.

**Comment by happytomatoe** [comment]
After short testing - 
I don't get error on this page anymore. I get this error on 
![image](https://github.com/user-attachments/assets/b0316484-8958-4eb5-a2c0-aad6603c41b4)
or when opening https://nand2tetris.github.io/web-ide/util

**Comment by DavidSouther** [comment]
I am able to open and run the VM Emulator in Firefox. Please re-open if you can provide specific steps to reproduce the issue.

<img width="1444" alt="image" src="https://github.com/user-attachments/assets/86637cc4-44aa-4840-b597-5563b7546ee0">
